### PR TITLE
[Aurora Code Scanning Fix] Add escape-html to req headers, protocol and file name

### LINKED ISSUE #2042: Fix code scanning alert - Server crash
Tracking issue for:
- [ ] https://github.com/Recidiviz/pulse-dashboard/security/code-scanning/4

### LINKED ISSUE #2043: Fix code scanning alert - Uncontrolled data used in path expression
Tracking issue for:
- [ ] https://github.com/Recidiviz/pulse-dashboard/security/code-scanning/2

### LINKED ISSUE #2044: Fix code scanning alert - Reflected cross-site scripting
Tracking issue for:
- [x] https://github.com/Recidiviz/pulse-dashboard/security/code-scanning/3

Related to [#460](https://github.com/Recidiviz/zenhub-tasks/issues/460)

### PR DESCRIPTION
**Pull request by HarryZ10**
## Description of the change

> This PR resolves the coding alert found in #2042, #2043, and #2044, adding the `escape-html` package to sanitize request headers, protocol, and the file names.

## Type of change

- [ ] Bug fix (non-breaking change that fixes an issue)
- [ ] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] Configuration change (adjusts configuration to achieve some end related to functionality, development, performance, or security)

## Related issues

> Closes #2042 
> Closes #2043 
> Closes #2044
> Closes [#460](https://github.com/Recidiviz/zenhub-tasks/issues/460)

### Tracking alert that does not have its own issue:
- [x] https://github.com/Recidiviz/pulse-dashboard/security/code-scanning/1

### Development

These boxes should be checked by the submitter prior to merging:

- [ ] Manual testing against realistic data has been performed locally
- [ ] E2E have been run for Lantern changes ([Steps in the Readme](https://github.com/Recidiviz/pulse-dashboard#running-e2e-tests))

### Code review

These boxes should be checked by reviewers prior to merging:

- [x] This pull request has a descriptive title and information useful to a reviewer
- [x] This pull request has been moved out of a Draft state, has no "Work In Progress" label, and has assigned reviewers
- [x] Potential security implications or infrastructural changes have been considered, if relevant

### CODE REVIEW

**Comment by phenggeler** [comment]
Thanks @HarryZ10 - I'm not comfortable approving without @Catacola, @jovergaag or @macfarlandian's input

File: server/routes/api.js [hunk 1]
```diff
@@ -203,17 +206,26 @@ function pathways(req, res) {
...
+
+      res.status(500);
+      res.end();
     }
   });
-  res.send(`${protocol}://${req.headers.host}/file/${fileName}`);
+
+  res.send(`${protocol}://${reqHeadersHost}/file/${fileName}`);
+  res.status(200);
+  res.end();
```
Comment (macfarlandian) [inline server/routes/api.js:228]: I don't think these lines are really doing anything? `res.send` is already sending a 200 and ending the request

File: server/routes/api.js [hunk 2]
```diff
@@ -203,17 +206,26 @@ function pathways(req, res) {
 
 function generateFileLink(req, res) {
   const { file } = req;
-  const fileName = `${uuid.v4()}-${file.originalname}`;
-  const protocol = process.env.AUTH_ENV === "development" ? `http` : `https`;
+  const fileName = sanitizeFilename(`${uuid.v4()}-${file.originalname}`);
+  const protocol =
+    escape(process.env.AUTH_ENV) === "development" ? `http` : `https`;
```
Comment (macfarlandian) [inline server/routes/api.js:211]: I don't think you actually need to escape the env var here? it's not user-supplied and it's not being rendered anywhere

File: server/routes/api.js [hunk 3]
```diff
@@ -203,17 +206,26 @@ function pathways(req, res) {
...
+
+  const reqHeadersHost = escape(req.headers.host);
 
   fs.writeFile(`/tmp/${fileName}`, file.buffer, function (err) {
     if (err) {
-      throw new Error(
```
Comment (macfarlandian) [inline server/routes/api.js:211]: this server has a global error handler configured [here](https://github.com/Recidiviz/pulse-dashboard/blob/main/server/app.js#L129) so I think the code-scanning alert related to this was actually a false positive ... it shouldn't be necessary to handle errors this way in this application

**Review by macfarlandian — CHANGES_REQUESTED** [review]

File: server/routes/api.js (hunk 3 above)
Comment (HarryZ10) [reply server/routes/api.js:211]: Confirming receipt and will resolve

File: server/routes/api.js (hunk 1 above)
Comment (HarryZ10) [reply server/routes/api.js]: fixed

File: server/routes/api.js [hunk 4]
```diff
@@ -203,17 +206,19 @@ function pathways(req, res) {
 
 function generateFileLink(req, res) {
   const { file } = req;
-  const fileName = `${uuid.v4()}-${file.originalname}`;
+  const fileName = `${uuid.v4()}-${sanitizeFilename(file.originalname)}`;
```
Comment (HarryZ10) [inline server/routes/api.js:209]: More info on: https://www.npmjs.com/package/sanitize-filename

**Review by macfarlandian — APPROVED** [review]
nice! thanks for addressing these

**Pull request merged**